File: bff/transport/rest/misc/info.go
package misc

import (
	"github.com/gin-gonic/gin"
)

var (
	version   = "dev"
	commit    = "n/a"
	buildDate = "n/a"
)

func Info(c *gin.Context) {
	c.JSON(200, gin.H{
		"version": version,
		"commit":  commit,
		"build":   buildDate,
	})
}

File: web/src/services/info.ts
import useSWR, { preload } from 'swr'

export interface Info {
  build: string
  commit: string
  version: string
}

const INFO_KEY = 'info'

const infoFetcher = async (): Promise<Info> => {
  const response = await fetch('/web/info')
  return response.json() as Promise<Info>
}

export function useInfo() {
  const { data, error, isLoading } = useSWR<Info, Error>(INFO_KEY, infoFetcher)
  return { data, error, isLoading }
}

export function getInfo(): Promise<Info> {
  return Promise.resolve(preload<Info>(INFO_KEY, infoFetcher))
}
